src/bases/finitebasis.jl

```diff
@@ -0,0 +1,60 @@
+import Base: length,
+    size
+
+###############
+# FiniteBasis #
+###############
+    # A FiniteBasis repesents the tensor product
+    # of an arbitrary number of fock bases, each of which is truncated 
+    # at a known, finite number of states. For example, imagine the 
+    # following fock bases and their tensor product:
+    #
+    # F_4 => {|0>, |1>, |2>, |3>}
+    # F_3 => {|0>, |1>, |2>}
+    # F_2 => {|0>, |1>} 
+    #
+    # F_4 x F_3 x F_2 => {|0,0,0>, |0,0,1>, |0,1,0>, |0,1,1>...|3,2,1>}
+    # 
+    # Note that the subscript number refers to basis *length*, not max
+    # excitation level.
+    #
+    # This tensor product basis, F_4 x F_3 x F_2, could be represented 
+    # by calling FiniteBasis(4, 3, 2). 
+
+    immutable FiniteBasis{S} <: AbstractBasis{S}
```

Comment:
I can't think of any significant performance issues with using a tuple type here - what did you have in mind?

Unless we are explicitly using `NTuple` as a field, or some similarly typed field, I'm weary of making `N` a parameter of `FiniteBasis` for the type stability issues I've previously discussed. Are you using `N` as the number of states, or the number of factor bases? The former I'm against, because that would be an abuse of the type system/probably lead to type stability issues - it would be like storing `length` as a type parameter for `Array`. I'm more open-minded but still reluctant about the latter.